generators/typescript/utils/core-utilities/fetcher/src/stream/Stream.ts

```diff
@@ -0,0 +1,134 @@
+import { Readable } from "stream";
+import { RUNTIME } from "../runtime";
+
+export declare namespace Stream {
+    interface Args {
+        /**
+         * The HTTP response stream to read from.
+         */
+        stream: Readable | ReadableStream;
+        /**
+         * The event shape to use for parsing the stream data.
+         */
+        eventShape: JsonEvent | SseEvent;
+    }
+
+    interface JsonEvent {
+        type: "json";
+        messageTerminator: string;
+    }
+
+    interface SseEvent {
+        type: "sse";
+        streamTerminator?: string;
+    }
+}
+
+const DATA_PREFIX = "data:";
```

Comment:
nit: Thoughts on `SSE_DATA_PREFIX`? At first I was concerned this was accidentally being used for the original JSON streaming approach.

Reply:
sounds great